### DISCUSSION TITLE
MUI TextField value overwritten by defaultValues in V7

### DISCUSSION THREAD

**Discussion by dslovinsky** [react-hook-form/react-hook-form#5666] [Q&A] [answered]
Hello,

I've encountered an issue when migrating to V7 in which values from MUI's TextFields are not getting registered properly when a default value is set in `useForm`.

Here's a simple [codesandbox](https://codesandbox.io/s/defaults-overwriting-textfield-xympi?file=/src/App.tsx) that demonstrates the problem.

As you can see, the values from the `<input>` element and `<TextField>` without a default get registered correctly and are logged to the console. However, the `<TextField>` with a default always logs the default value regardless of user input.

At the moment I've solved this in the application with `useController`, but it would be preferable if the value could be retrieved without that, especially given that doing so worked as normal prior to V7.

**Comment by bluebill1049** [comment, marked as answer]
This is due to `ref` name as `inputRef`.  while register is return `ref` i would recommend to use useController/Controller for MUI

**Reply by dslovinsky** [reply]
Alright, thanks for the quick response!

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api/usecontroller/controller

https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5

this CSB above shoud help.
